client/platform/desktop/frontend/components/MultiTrainingMenu.vue

```diff
@@ -0,0 +1,236 @@
+<script lang="ts">
+import type { DataTableHeader } from 'vuetify';
+
+import {
+  computed, defineComponent, onBeforeMount, set, del, reactive,
+} from '@vue/composition-api';
+import { DatasetMeta, TrainingConfigs, useApi } from 'viame-web-common/apispec';
+import { datasets } from '../store/dataset';
+
+export default defineComponent({
+  setup(_, { root }) {
+    const { getTrainingConfigurations, runTraining } = useApi();
+
+    const data = reactive({
+      stagedItems: {} as Record<string, DatasetMeta>,
+      trainingOutputName: '',
+      selectedTrainingConfig: 'foo.whatever',
+      trainingConfigurations: {
+        configs: [],
+        default: '',
+      } as TrainingConfigs,
+    });
+
+    const headersTmpl: DataTableHeader[] = [
+      {
+        text: 'Dataset',
+        value: 'name',
+        sortable: true,
+      },
+      {
+        text: 'Type',
+        value: 'type',
+        sortable: true,
+        width: 160,
+      },
+      {
+        text: 'fps',
+        value: 'fps',
+        sortable: true,
+        width: 80,
+      },
+    ];
+
+    onBeforeMount(async () => {
+      const configs = await getTrainingConfigurations();
+      data.trainingConfigurations = configs;
+      data.selectedTrainingConfig = configs.default;
+    });
+
+    function toggleStaged(meta: DatasetMeta) {
+      if (data.stagedItems[meta.id]) {
+        del(data.stagedItems, meta.id);
+      } else {
+        set(data.stagedItems, meta.id, meta);
+      }
+    }
+
+    const availableItems = computed(() => Object.values(datasets.value)
+      .map((item) => ({
+        ...item,
+        included: item.id in data.stagedItems,
+      })));
+
+    const stagedItems = computed(() => Object.values(data.stagedItems));
+
+    const isReadyToTrain = computed(() => (
+      stagedItems.value.length > 0
+        && data.selectedTrainingConfig
+        && data.trainingOutputName
+    ));
+
+    async function runTrainingOnFolder() {
```

Comment:
I don't know if it's my Windows PC (I don't think so) but there is a decent delay between starting a cmd exe and it running and switching to the jobs folder.  It can be like 3-5 seconds, enough time where I think I didn't click on button.  I think we may want some indication on the page that job has started the process.  Like the button being disabled and having a 'starting' or something else display.

Reply:
This is likely caused by `validateViamePath` being called before every job (sanity check).

It's instant on linux, so it costs nothing to perform.  But since windows is terrible and you had to add that progress thing on the settings page, maybe we should just disable it?

Reply:
See  915849e

Reply:
Correct, it seems to be the validation that slows it down.  We could do a onetime confirmation on first training/pipeline run.  Although I understand it may not be needed because without it the training will fail anyways if there is an issue.  I just don't know how helpful the error message will be.

I do something on first conversion for ffmpeg with linux where I need to store the location/version of ffmpeg so that subsequent conversions don't execute the decision logic of finding the proper ffmpeg  installed (VIAME x264, local, VIAME nv_enc)

Reply:
How long does that take?  Can it be done once lazily and held in memory, or do you really need to persist it?

Reply:
I doesn't take long but it's a cascade of `spawnSync`  processes (I want them to occur so I can respond in UI to the user if there is an error)  based on what versions it can find.  I only hold in memory  because really it will only be valid during the execution of the app.  User could install, uninstall, move or change stuff in between and not understand the connection.  If they do while the app is open I would hope they could see the connection as to why it failed, especially with relevant error messages.